### ISSUE TITLE
[clang-tidy] Warn about wrong inline keyword placement

### ISSUE THREAD

**Issue by SunBlack** [llvm/llvm-project#80686]
As  #72397 already mention, the [inline keyword](https://en.cppreference.com/w/cpp/language/inline) is widely misunderstood. I feel like I've already seen all possible variants where the keyword can be added.

The new check [readability-redundant-inline-specifier](https://clang.llvm.org/extra/clang-tidy/checks/readability/redundant-inline-specifier.html) already detects code like this:

```cpp
class A
{
public:
  inline int test(){
    return 0;
  }
};

int main()
{
    A a;
    return a.test();
};
```

But it doesn't warn in cases like this (I believe one is redundant - I think the inline before the implementation is better):
```cpp
class A
{
public:
  inline int test();
};

inline int A::test(){
    return 0;
}

int main()
{
    A a;
    return a.test();
};
```

Another missing warning: In case the implementation is within the cpp the `inline` has no effect:
```cpp
// a.h
class A
{
public:
  inline int test();
};

// a.cpp
#include "a.h"

inline int A::test(){
    return 0;
}
```
